### DISCUSSION TITLE
DeleteEvent option for SelfAggregate

### DISCUSSION THREAD

**Discussion by No9993** [JasperFx/marten#2498] [Q&A]
Add a delete option for projection type SelfAggregate to support Aggregates that can get deleted.

**Comment by jeremydmiller** [comment]
Or use `AggregateProjection<T>` instead. This is already supported for live, inline, or asynchronous projections. The Self-Aggregate gets compiled to an `AggregateProjection<T>` behind the scenes anyway. I'm converting this to a question, but there's no development action to take.

**Comment by dgrozenok** [comment]
I am using below for the self aggregates and the projected documents are deleted on `ProductDeleted` event.
```csharp
public bool ShouldDelete(ProductDeleted _) => true;
```
Am I missing anything?

**Reply by No9993** [reply]
Sorry, how exactly are you implementing this :) Can u provide the context?

**Reply by No9993** [reply]
nvm, just a bool func on the aggregate, brain fart. Thanks for the insight, works like a charm
